src-cpp/app/modules/shared/section_item.h

```diff
@@ -0,0 +1,118 @@
+#ifndef SECTION_ITEM_H
+#define SECTION_ITEM_H
+
+#include <QObject>
+#include <QString>
+
+namespace Shared
+{
+namespace Models
+{
+enum SectionType
+{
+    Chat = 0,
+    Community,
+    Wallet,
+    Browser,
+    ProfileSettings,
+    NodeManagement
+};
+class SectionItem: public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(QString id READ getId)
+    Q_PROPERTY(int sectionType READ getSectionType)
+    Q_PROPERTY(QString name READ getName)
+    Q_PROPERTY(bool amISectionAdmin READ getAmISectionAdmin)
+    Q_PROPERTY(QString description READ getDescription)
+    Q_PROPERTY(QString image READ getImage)
+    Q_PROPERTY(QString icon READ getIcon)
+    Q_PROPERTY(QString color READ getColor)
+    Q_PROPERTY(bool hasNotification READ getHasNotification)
+    Q_PROPERTY(int notificationsCount READ getNotificationsCount)
+    Q_PROPERTY(bool active READ getIsActive NOTIFY activeChanged)
+    Q_PROPERTY(bool enabled READ getIsEnabled)
+    Q_PROPERTY(bool joined READ getHasJoined)
+    Q_PROPERTY(bool isMember READ getIsMember)
```

Comment:
if you use `non-notify` properties you can use `member` in `Q_PROPERTY`

Reply:
aaah I just read that just now, thanks for pointing it out, however do you know how do I make it read only?
Q_PROPERTY(bool enabled MEMBER m_isEnabled), this would mean read and write both right?

Reply:
but that properties never changed?

Reply:
what do you mean?
if I pass Q_PROPERTY(bool enabled MEMBER m_isEnabled), someone on the QML side, using it could also assign values to this m_isEnabled right? and we dont want that